lib/rspec/rails/configuration.rb

```diff
@@ -45,15 +45,40 @@ class Configuration
     #
     # @api private
     def self.add_test_type_configurations(config)
-      config.include RSpec::Rails::ControllerExampleGroup, type: :controller
-      config.include RSpec::Rails::HelperExampleGroup,     type: :helper
-      config.include RSpec::Rails::ModelExampleGroup,      type: :model
-      config.include RSpec::Rails::RequestExampleGroup,    type: :request
-      config.include RSpec::Rails::RoutingExampleGroup,    type: :routing
-      config.include RSpec::Rails::ViewExampleGroup,       type: :view
-      config.include RSpec::Rails::FeatureExampleGroup,    type: :feature
+      config.define_derived_metadata(type: :controller) do
+        config.include RSpec::Rails::ControllerExampleGroup, type: :controller
+      end
+
+      config.define_derived_metadata(type: :helper) do
+        config.include RSpec::Rails::HelperExampleGroup, type: :helper
+      end
+
+      config.define_derived_metadata(type: :model) do
+        config.include RSpec::Rails::ModelExampleGroup, type: :model
+      end
+
+      config.define_derived_metadata(type: :request) do
+        config.include RSpec::Rails::RequestExampleGroup, type: :request
+      end
+
+      config.define_derived_metadata(type: :routing) do
+        require "action_controller/test_case"
```

Comment:
This is necessary because `RSpec::Rails::RoutingExampleGroup` depends on `ActionDispatch::Assertions::RoutingAssertions`, which references `ActionController::TestRequest` without requiring the file where it's defined:
https://github.com/rails/rails/blob/v7.1.3.2/actionpack/lib/action_dispatch/testing/assertions/routing.rb#L242

I've fixed that upstream in https://github.com/rails/rails/commit/53079761a55a7460c75db178bd48bf6b26588408, but for compatibility with all existing Rails versions we need to keep eager loading `ActionController::TestCase` here.